networkx/algorithms/maximum_weight_cycle_packing.py

```diff
@@ -0,0 +1,229 @@
+        #   adding the nodes in the graph
+        for edge in Y:
+            ans_graph.add_node((edge[0], edge[1]))
+            seen_Y.add(edge[0])
+            seen_Y.add(edge[1])
+            if (edge[0], edge[1]) in graph.edges and (edge[1], edge[0]) in graph.edges:
+                weight = (
+                    graph.get_edge_data(edge[0], edge[1])["weight"]
+                    + graph.get_edge_data(edge[1], edge[0])["weight"]
+                )
+                ans_graph.add_edge(
+                    (edge[0], edge[1]),
+                    (edge[0], edge[1]),
+                    weight=weight,
+                    cycle=[edge[0], edge[1]],
+                )
+        for edge in graph.edges:
+            if edge[0] not in seen_Y and edge[0] not in X:
+                X.append(edge[0])
+                ans_graph.add_node(edge[0])
+        connect_2cycles(X, graph, ans_graph)
+        connect_3cycles(X, Y, graph, ans_graph)
+        exchanges = list(nx.max_weight_matching(ans_graph))
+        if (
+            len(exchanges) == 0 and ans_graph.number_of_edges() == 1
+        ):  # for the use-case of only self connected edge
+            exchanges = [list(ans_graph.edges)[0]]
+        temp_max = 0
+        for cyc in exchanges:
+            temp_max = temp_max + ans_graph.get_edge_data(cyc[0], cyc[1])["weight"]
+        if temp_max > max_weight:
+            max_weight = temp_max
+            max_cycles = exchanges
+            max_graph = ans_graph.copy()
+
+    result = []  # exctract only the cycles
+    for cyc in max_cycles:
+        cycle = max_graph.get_edge_data(cyc[0], cyc[1])["cycle"]
+        result.append(cycle)
+
+    return result  # exchanges
+
+
+def connect_2cycles(X, graph, ans_graph):
+    for i in range(
+        len(X)
+    ):  # creating the edges in the graph by going through the 2-circles
+        for j in range(i + 1, len(X)):
+            if (X[i], X[j]) in graph.edges and (X[j], X[i]) in graph.edges:
+                weight = (
+                    graph.get_edge_data(X[i], X[j])["weight"]
+                    + graph.get_edge_data(X[j], X[i])["weight"]
+                )
+                ans_graph.add_edge((X[i]), (X[j]), weight=weight, cycle=[X[i], X[j]])
+
+
+def connect_3cycles(X, Y, graph, ans_graph):
+    #   creating the edges in the graph by going through the 3-circles
+    for k in range(len(X)):
+        for j, l in Y:  # This deals with the normal case of Yi,j Xk
+            if (l, X[k]) in graph.edges and (
+                X[k],
+                j,
+            ) in graph.edges:  # [j, l, X[k]] in cycles:
+                weight = (
+                    graph.get_edge_data(j, l)["weight"]
+                    + graph.get_edge_data(l, X[k])["weight"]
+                    + graph.get_edge_data(X[k], j)["weight"]
+                )
+                ans_graph.add_edge((X[k]), (j, l), weight=weight, cycle=[j, l, X[k]])
+
+
+def simple_cycles(G, limit):
+    """t
```

Comment:
Add documentation to this function.

Reply:
Also, since this function is useful for both your algorithms, I would put it in a separate file, for example  `algorithms/simple_cycles.py`, so that you can import and use it in both algorithms.